src/DotVVM.Core/Controls/Options/SortingOptions.cs

```diff
@@ -22,8 +22,9 @@ public class SortingOptions : ISortingOptions
         public string SortExpression { get; set; }
```

Comment:
It's quite unfortunate that this has to be a `string`, but what about to validate it the property exists in the setter.

Reply:
The expression is validated before `IQueryable` is being sorted. But we need to support sorting by multiple expressions. I would rather change this API, but I am not sure it's good to introduce another breaking change here.

Reply:
Hmm, it's just a string, we may introduce special format for multiple expressions...

Reply:
I was thinking about adding MultiCriteriaSortingOptions with a collection of additional sort expressions (and directions). Majority of users don't need it, so they can have null in this property.

Reply:
There are several options: 
1. Use SortingOptions for the first criteria and MultiCriteriaSortingOptions for the other ones (which would allow to use the DataSet even in controls which do not support multi-criteria sorting). We can create extension methods to work with the sort criteria in an unified way. 
2. Use MultiCriteriaSortingOptions for all criteria. In that case, SortingOptions must be null.
3. Or we might add AdditionalSortCriteria collection right in the SortingOptions and forget about MultiCriteriaSortingOptions. 

The first way seems unnecessarily complicated - I would go for 2 or 3. Option 3 means breaking the ISortingOptions interface, however I wouldn't mind so much - I guess nobody except us implemented that.

Reply:
Extending the `ISortOptions` interface (3) seems the most sane option for me.

Btw, @tomasherceg you wanted a example where interface does not help with extensibility :P

Reply:
OOP sucks, go functional!

Reply:
@tomasherceg So, finally? :P